### PR TITLE
[FLINK-23427][chinese-translation] Translate the page of "Blocking Sh…

### PR DESCRIPTION
**Pull request by GOODBOY008**
…uffle" into Chinese

## What is the purpose of the change

Translate the internal page "https://ci.apache.org/projects/flink/flink-docs-master/zh/docs/ops/batch/blocking_shuffle/" into Chinese.

The doc located in "flink/docs/content.zh/docs/ops/batch/blocking_shuffle.md"

## Brief change log

Translate  the doc located in "flink/docs/content.zh/docs/ops/batch/blocking_shuffle.md"

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no )
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no )
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by GOODBOY008** [comment]
@wuchong Can you review my pr? And give me some suggestions.

**Comment by wuchong** [comment]
@gaoyunhaii could you help to review this? I think you should be familiar with this part as you contributed the original English docs.

**Comment by gaoyunhaii** [comment]
No problem, I'll have a look~

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 1]
```diff
@@ -27,63 +27,63 @@ under the License.
 
 # Blocking Shuffle
 
-## Overview
+## 总览
 
-Flink supports a batch execution mode in both [DataStream API]({{< ref "docs/dev/datastream/execution_mode" >}}) and [Table / SQL]({{< ref "/docs/dev/table/overview" >}}) for jobs executing across bounded input. In this mode, network exchanges occur via a blocking shuffle. Unlike the pipeline shuffle used for streaming applications, blocking exchanges persists data to some storage. Downstream tasks then fetch these values via the network. Such an exchange reduces the resources required to execute the job as it does not need the upstream and downstream tasks to run simultaneously.
+Flink 在使用 [DataStream API]({{< ref "docs/dev/datastream/execution_mode" >}}) 和 [Table / SQL]({{< ref "/docs/dev/table/overview" >}}) 都支持批处理执行模式，这用于任务处理有界输入流。使用此模式通过 blocking shuffle 来网络传输。不同于在流式应用使用管道 shuffle，阻塞交换的数据并存储.然后下游任务通过网络获取这些值。这种交换减少了执行作业所需的资源，因为它不需要同时运行上游和下游任务。
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:32]: 1. <del>在使用</del>；
2. “都支持批处理执行模式，这用于任务处理有界输入流” -> “都支持通过批处理执行模式处理有界输入”；
3. ”使用此模式通过 blocking shuffle 来网络传输“，”阻塞交换的数据并存储“，缺乏主语且语句不通顺。请修改。
4. 含有英文句点“.”，请注意格式。

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 2]
```diff
@@ -27,63 +27,63 @@ under the License.
...
+## 总览
 
-Flink supports a batch execution mode in both [DataStream API]({{< ref "docs/dev/datastream/execution_mode" >}}) and [Table / SQL]({{< ref "/docs/dev/table/overview" >}}) for jobs executing across bounded input. In this mode, network exchanges occur via a blocking shuffle. Unlike the pipeline shuffle used for streaming applications, blocking exchanges persists data to some storage. Downstream tasks then fetch these values via the network. Such an exchange reduces the resources required to execute the job as it does not need the upstream and downstream tasks to run simultaneously.
+Flink 在使用 [DataStream API]({{< ref "docs/dev/datastream/execution_mode" >}}) 和 [Table / SQL]({{< ref "/docs/dev/table/overview" >}}) 都支持批处理执行模式，这用于任务处理有界输入流。使用此模式通过 blocking shuffle 来网络传输。不同于在流式应用使用管道 shuffle，阻塞交换的数据并存储.然后下游任务通过网络获取这些值。这种交换减少了执行作业所需的资源，因为它不需要同时运行上游和下游任务。 
 
-As a whole, Flink provides two different types of blocking shuffles; `Hash shuffle` and `Sort shuffle`.
+总的来说，Flink 提供了两种不同类型的 blocking shuffles ; `Hash shuffle` 和 `Sort shuffle`。
 
-They will be detailed in the following sections.
+在下面章节会详细说明它们.
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:36]: 中文翻译不应含有英文句点。

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 3]
```diff
@@ -27,63 +27,63 @@ under the License.
...
 
 ## Hash Shuffle
 
-The default blocking shuffle implementation, `Hash Shuffle`, has each upstream task persist its results in a separate file for each downstream task on the local disk of the TaskManager. When the downstream tasks run, they will request partitions from the upstream TaskManager's, which read the files and transmit data via the network.
+`Hash Shuffle` 是 blocking shuffle 的默认实现，它为每个下游任务将每个上游任务的结果以单独文件的方式保存在 TaskManager 本地磁盘上。当下游任务运行时会向上游的 TaskManager 请求分片，TaskManager 读取文件之后通过网络传输。
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:40]: 通过网络传输（给下游任务）。

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 4]
```diff
@@ -27,63 +27,63 @@ under the License.
...
 
-`Hash Shuffle` provides different mechanisms for writing and reading files:
+`Hash Shuffle` 为读写文件提供了不同的机制:
 
-- `file`: Writes files with the normal File IO, reads and transmits files with Netty `FileRegion`. `FileRegion` relies on `sendfile` system call to reduce the number of data copies and memory consumption.
-- `mmap`: Writes and reads files with `mmap` system call.
-- `Auto`: Writes files with the normal File IO, for file reading, it falls back to normal `file` option on 32 bit machine and use `mmap` on 64 bit machine. This is to avoid file size limitation of java `mmap` implementation on 32 bit machine.
+- `file`: 通过标准文件 IO 写文件，读取和传输文件是要通过 Netty 的 `FileRegion`。`FileRegion` 依靠系统调用 `sendfile` 来减少数据拷贝和内存消耗。
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:44]: 是要 -> 需要

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 5]
```diff
@@ -27,63 +27,63 @@ under the License.
...
+- `mmap`: 通过系统调用 `mmap` 来读写文件。
+- `Auto`: 通过标准文件 IO 写文件，对于文件读取，在32为机器上降级到 `file` 选项 并且 在64位机器上使用 `mmap` 。这是为了避免在 32 位机器上 java 实现 `mmap` 的文件大小限制。
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:46]: 为 -> 位。

数字和中文之间需要添加空格，详见[《中文文案排版指北》](https://github.com/sparanoid/chinese-copywriting-guidelines/blob/master/README.zh-CN.md#%E4%B8%AD%E6%96%87%E4%B8%8E%E6%95%B0%E5%AD%97%E4%B9%8B%E9%97%B4%E9%9C%80%E8%A6%81%E5%A2%9E%E5%8A%A0%E7%A9%BA%E6%A0%BC)。

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 6]
```diff
@@ -27,63 +27,63 @@ under the License.
...
+如果开启 [SSL]({{< ref "docs/deployment/security/security-ssl" >}})，`file` 机制不能使用 `FileRegion` 而是在传输之前使用非池化的缓存去缓存数据。这可能会 [cause direct memory OOM](https://issues.apache.org/jira/browse/FLINK-15981)。 此外，因为同步读取文件有时会造成 netty 线程阻塞，[SSL handshake timeout]({{< ref "docs/deployment/config#security-ssl-internal-handshake-timeout" >}}) 配置需要上调以防 [connection reset errors](https://issues.apache.org/jira/browse/FLINK-21416)。
 {{< /hint >}}
 
 {{< hint info >}}
-The memory usage of `mmap` is not accounted for by configured memory limits, but some resource frameworks like Yarn will track this memory usage and kill the container if memory exceeds some threshold.
+`mmap`使用的内存不计算进已有配置的内存限制中， 但是一些资源管理框架比如 yarn 将追踪这块内存使用并且如果容器使用内存超过阈值会被杀掉。
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:59]: ”并且“前添加逗号。

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 7]
```diff
@@ -27,63 +27,63 @@ under the License.
...
 {{< /hint >}}
 
-To further improve the performance, for most jobs we also recommend [enabling compression]({{< ref "docs/deployment/config">}}#taskmanager-network-blocking-shuffle-compression-enabled) unless the data is hard to compress.
+为了进一步的提升性能， 对于绝大多数的任务我们推荐 [enabling compression]({{< ref "docs/deployment/config">}}#taskmanager-network-blocking-shuffle-compression-enabled) 除非数据很难被压缩。
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:62]: 逗号后多余空格，请注意格式。

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 8]
```diff
@@ -27,63 +27,63 @@ under the License.
...
 
-`Hash Shuffle` works well for small scale jobs with SSD, but it also have some disadvantages:
+`Hash Shuffle` 在小规模运行在固态硬盘的任务情况下效果显著，但是依旧有一些挑战:
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:64]: “挑战” -> 问题。

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 9]
```diff
@@ -27,63 +27,63 @@ under the License.
...
 
-1. If the job scale is large, it might create too many files, and it requires a large write buffer to write these files at the same time.
-2. On HDD, when multiple downstream tasks fetch their data simultaneously, it might incur the issue of random IO.
+1. 如果任务的规模庞大将会创建很多文件并且要求同时对这些文件进行大量的写操作。
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:66]: 长句中添加逗号。

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 10]
```diff
@@ -27,63 +27,63 @@ under the License.
...
+`Sort Shuffle` 是 1.13 版中引入的另一种 blocking shuffle 实现.不同于 `Hash Shuffle`，sort shuffle 将每个分区结果写入到一个文件。当多个下游任务同时读取结果分片，数据文件只会被打开一次并共享给所有的读请求。总之，集群使用更少的资源例如：节点和文件描述符以提升稳定性。此外，通过写更少的文件和尽可能线性的读取文件，尤其是在使用机械硬盘情况下 sort shuffle 可以获得比 hash shuffle 更好的性能。另外，`sort shuffle` 使用额外管理的内存作为读数据缓存并不依赖 `sendfile` 或 `mmap` 机制， 因此也适用于 [SSL]({{< ref "docs/deployment/security/security-ssl" >}})。关于 sort shuffle 的更多细节请参考 [FLINK-19582](https://issues.apache.org/jira/browse/FLINK-19582) 和 [FLINK-19614](https://issues.apache.org/jira/browse/FLINK-19614) .
```
Comment (tsreaper) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:71]: 总之？

例如前添加句号。

逗号后多余空格，请注意格式。

**Review by tsreaper — CHANGES_REQUESTED** [review]
Thanks for contributing to Flink! However there are quite some issues in this translation, especially on the format and on the fluency of the language. I've made some comments.

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md (hunk 10 above)
Comment (GOODBOY008) [reply docs/content.zh/docs/ops/batch/blocking_shuffle.md:71]: "总之"->"因此"。

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md (hunk 1 above)
Comment (GOODBOY008) [reply docs/content.zh/docs/ops/batch/blocking_shuffle.md:32]: 3 ”使用此模式通过 blocking shuffle 来网络传输“->"此模式是通过 blocking shuffle 进行网络传输。 "

Comment (GOODBOY008) [reply docs/content.zh/docs/ops/batch/blocking_shuffle.md:32]: "不同于在流式应用使用管道 shuffle，阻塞交换的数据并存储.然后下游任务通过网络获取这些值。"->"与流式应用使用管道 shuffle 阻塞交换的数据并存储，然后下游任务通过网络获取这些值的方式不同。"

**Comment by GOODBOY008** [comment]
@gaoyunhaii Please review my pr again.Thank you.

**Comment by gaoyunhaii** [comment]
Very thanks for the update! I'll have a look as soon as possible~

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 11]
```diff
@@ -27,63 +27,63 @@ under the License.
...
 # Blocking Shuffle
 
-## Overview
+## 总览
 
-Flink supports a batch execution mode in both [DataStream API]({{< ref "docs/dev/datastream/execution_mode" >}}) and [Table / SQL]({{< ref "/docs/dev/table/overview" >}}) for jobs executing across bounded input. In this mode, network exchanges occur via a blocking shuffle. Unlike the pipeline shuffle used for streaming applications, blocking exchanges persists data to some storage. Downstream tasks then fetch these values via the network. Such an exchange reduces the resources required to execute the job as it does not need the upstream and downstream tasks to run simultaneously.
+Flink [DataStream API]({{< ref "docs/dev/datastream/execution_mode" >}}) 和 [Table / SQL]({{< ref "/docs/dev/table/overview" >}}) 都支持通过批处理执行模式处理有界输入。此模式是通过 blocking shuffle 进行网络传输。与流式应用使用管道 shuffle 阻塞交换的数据并存储，然后下游任务通过网络获取这些值的方式不同。这种交换减少了执行作业所需的资源，因为它不需要同时运行上游和下游任务。 
 
-As a whole, Flink provides two different types of blocking shuffles; `Hash shuffle` and `Sort shuffle`.
+总的来说，Flink 提供了两种不同类型的 blocking shuffles ; `Hash shuffle` 和 `Sort shuffle`。
```
Comment (gaoyunhaii) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:34]: `;` -> `:` ?

File: docs/content.zh/docs/ops/batch/blocking_shuffle.md [hunk 12]
```diff
@@ -27,63 +27,63 @@ under the License.
...
 {{< hint info >}}
-Currently `sort shuffle` only sort records by partition index instead of the records themselves, that is to say, the `sort` is only used as a data clustering algorithm.
+目前 `sort shuffle` 只通过分区索引来排序而不是记录本身，也就是说 `sort` 只是被当成数据聚类算法使用。
 {{< /hint >}}
 
-## Choices of Blocking Shuffle
+## 如何选择 Blocking Shuffle
 
-As a summary,
+总结，
```
Comment (gaoyunhaii) [inline docs/content.zh/docs/ops/batch/blocking_shuffle.md:85]: 总结 -> 一般来说  或者  总的来说？

**Review by gaoyunhaii — COMMENTED** [review]
Hi @GOODBOY008 very sorry for the extreme long delay for the review for get bounded in the previous days. I only have two minor comments now.

**Comment by GOODBOY008** [comment]
@gaoyunhaii Thanks for your review,I have apply your suggestion.Please cc.

File: docs/content/docs/ops/batch/blocking_shuffle.md [hunk 13]
```diff
@@ -31,7 +31,7 @@ under the License.
 
 Flink supports a batch execution mode in both [DataStream API]({{< ref "docs/dev/datastream/execution_mode" >}}) and [Table / SQL]({{< ref "/docs/dev/table/overview" >}}) for jobs executing across bounded input. In this mode, network exchanges occur via a blocking shuffle. Unlike the pipeline shuffle used for streaming applications, blocking exchanges persists data to some storage. Downstream tasks then fetch these values via the network. Such an exchange reduces the resources required to execute the job as it does not need the upstream and downstream tasks to run simultaneously.
 
-As a whole, Flink provides two different types of blocking shuffles; `Hash shuffle` and `Sort shuffle`.
+As a whole, Flink provides two different types of blocking shuffles: `Hash shuffle` and `Sort shuffle`.
```
Comment (gaoyunhaii) [inline docs/content/docs/ops/batch/blocking_shuffle.md:34]: Hi @GOODBOY008 ~ I think we do not need to modify the english version~

**Comment by gaoyunhaii** [comment]
Hi @GOODBOY008 very thanks for opening the PR! LGTM % one above small comment. If you also agree with that, I'll merge the PR without that modification~

File: docs/content/docs/ops/batch/blocking_shuffle.md (hunk 13 above)
Comment (GOODBOY008) [reply docs/content/docs/ops/batch/blocking_shuffle.md:34]: ok,I fix it now.

**Review by gaoyunhaii — APPROVED** [review]

**Pull request closed without merging**